diff -rupN cobbler-2.0.7.old/koan/app.py cobbler-2.0.7/koan/app.py
--- cobbler-2.0.7.old/koan/app.py.org    2016-03-15 12:58:22.739224509 -0400
+++ cobbler-2.0.7/koan/app.py            2016-03-15 12:58:49.898224509 -0400
@@ -922,6 +922,17 @@ class Koan:

             utils.subprocess_call(cmd)

+            # Need to remove the root= argument to prevent booting the current OS
+            cmd = [
+                "/sbin/grubby",
+                "--update-kernel",
+                self.safe_load(
+                    profile_data,
+                    'kernel_local'),
+                "--remove-args=root"]
+
+            utils.subprocess_call(cmd)
+
             # Any post-grubby processing required (e.g. ybin, zipl, lilo)?
             if not ANCIENT_PYTHON and arch.startswith("ppc") and "grub2" not in probe_output:
                 # FIXME - CHRP hardware uses a 'PPC PReP Boot' partition and doesn't require running ybin
