diff -rupN cobbler-2.0.7.org/koan/utils.py cobbler-2.0.7/koan/utils.py
--- cobbler-2.0.7.org/koan/utils.py	2015-10-09 08:27:48.306630477 -0400
+++ cobbler-2.0.7/koan/utils.py	2015-10-09 08:27:11.026630477 -0400
@@ -213,7 +213,7 @@ def input_string_or_hash(options,delim=N
         new_dict = {}
         tokens = string.split(options, delim)
         for t in tokens:
-            tokens2 = string.split(t,"=")
+            tokens2 = string.split(t,"=",1)
             if len(tokens2) == 1:
                 # this is a singleton option, no value
                 key = tokens2[0]
